### ISSUE TITLE
[Dialog] Improve assistive technology (AT) experience for `dragEnabled` and `resizable`

### ISSUE THREAD

**Issue by geospatialem** [Esri/calcite-design-system#10229]
### Check existing issues

- [X] I have [checked for existing issues](https://github.com/Esri/calcite-design-system/issues) to avoid duplicates

### Summary

The component provides _some_ support to assistive technology (AT) users, but its unclear of the support since they must enter via forms mode to access the component, and they may enable the functionality without knowing and will not receive context of their changes.

### Actual Behavior

In forms mode AT users may activate `dragEnabled` using the arrow keys unknowingly. Users are not currently provided context of the changes made for both `dragEnabled` and `resizable`, so they could be unaware of any changes taking place.

### Expected Behavior

- [x] Remove `aria-grabbed` due to its deprecation and inconsistent support
- [ ] Remove dragging via arrows by default. Instead **_offer a method to enter or leave the drag mode_**, such as a button to enter, listen for the keystrokes, respond, then Escape or Space to exit. 
    - This was the recommendation from Level Access, because a screen reader user will get to the component and try to read it. If they are in Forms mode which allows key presses to go straight through the screen reader, they will be moving the component around instead of reading it.
- [ ] Provide context when interacting with the component's drag and resizing utilities via a live region. **_Nice to have: Provide context if at  the edge of the viewport._**

### Reproduction Sample

https://codepen.io/geospatialem/pen/BagmoWa

### Reproduction Steps

1. Enable either JAWS or NVDA
2. Open the [sample](https://codepen.io/geospatialem/pen/BagmoWa)
3. When the component is in focus turn on forms mode
    - JAWS: `Enter` key
    - NVDA: `Inst` + `spacebar` keys
4. Observe that dragging (arrow keys) and resizing (`shift` + arrow keys) works, but its not intuitive after entering form mode nor is context provided to the user

### Reproduction Version

2.12.0

### Working W3C Example/Tutorial

https://www.nngroup.com/articles/drag-drop/#toc-accessibility-for-draganddrop-4

### Relevant Info

Confirmed with JAWS and NVDA on Windows browsers

### Priority impact

impact - p2 - want for an upcoming milestone

### Calcite package

- [X] @esri/calcite-components
- [ ] @esri/calcite-components-angular
- [ ] @esri/calcite-components-react
- [ ] @esri/calcite-design-tokens
- [ ] @esri/eslint-plugin-calcite-components

### Esri team

Calcite (dev)

**monday.com sync:** #9366785390

**Comment by geospatialem** [comment]
Adding `design` for expertise on the following:

> Remove dragging via arrows by default. Instead offer a method to enter or leave the drag mode, such as a button to enter, listen for the keystrokes, respond, then Escape or Space to exit.

**Comment by driskull** [comment]
> Remove aria-grabbed due to its deprecation and inconsistent support

We never added this so ✅ 

> Nice to have: Provide context if at the edge of the viewport.

This one would be a bit tough to do.

**Comment by chezHarper** [comment]
Okay, designer Keith here.
So I've done some NVDA testing on dialogue and, as I tab through, the screen reader calls out both draggable capabilities( arrow keys) and resize (option+arrow keys); however, i can only get the drag working, not the resize.
Check it out: https://esriis-my.sharepoint.com/:v:/r/personal/kei14429_esri_com/Documents/A11y%20test_dragable%26resizable_dialogue.mp4?csf=1&web=1&e=SiVZnb&nav=eyJyZWZlcnJhbEluZm8iOnsicmVmZXJyYWxBcHAiOiJTdHJlYW1XZWJBcHAiLCJyZWZlcnJhbFZpZXciOiJTaGFyZURpYWxvZy1MaW5rIiwicmVmZXJyYWxBcHBQbGF0Zm9ybSI6IldlYiIsInJlZmVycmFsTW9kZSI6InZpZXcifX0%3D

**Comment by chezHarper** [comment]
I see what you're saying. I propose non-visible elements that can be tabbed to and controlled with the arrow keys for both the drag and the resize. What do you think of that?

<img width="625" height="224" alt="Image" src="https://github.com/user-attachments/assets/ea42d411-b9ee-4444-856d-0990759ed08c" />

<img width="625" height="224" alt="Image" src="https://github.com/user-attachments/assets/def5351c-fb8f-439f-91c5-d1048fddb92a" />

**Comment by driskull** [comment]
Yes that would work. It would still need focus styling but only shown when accessed via keyboard. So the main design question is what are the dimensions and placement of it.

**Comment by geospatialem** [comment]
For the development of the above, we should seek a prototype where we can get some regular assistive technology (AT) users (third party) testing to ensure the UI/UX is meeting the needs for more audiences, with folks that rely on AT daily.

**Once completed lets coordinate on a path forward in coordination with @chezHarper and I.**

**Comment by chezHarper** [comment]
@driskull I'll nail down placement and dimensions of resizing and dragging elements (and if they should be single or separate spans/divs).

**Comment by geospatialem** [comment]
Updating the above as `p - high` as this surfaced as part of the 2025 third party accessibility audit.
